src/api/app/models/user_ldap_strategy.rb

```diff
@@ -339,50 +339,49 @@ def self.initialize_ldap_con(user_name, password)
     return unless defined?(CONFIG['ldap_servers'])
     require 'ldap'
     ldap_servers = CONFIG['ldap_servers'].split(":")
-    count = 0
 
     max_ldap_attempts = CONFIG.has_key?('ldap_max_attempts') ? CONFIG['ldap_max_attempts'] : 10
 
-    while count < max_ldap_attempts
-      count += 1
+    # Do 10 attempts to connect to one of the configured LDAP servers. LDAP server
+    # to connect to is chosen randomly.
+    (max_ldap_attempts || 10).times do
       server = ldap_servers[rand(ldap_servers.length)]
       conn = try_ldap_con(server, user_name, password)
-      if !conn.nil? && conn.bound?
-        return conn
-      end
+
+      return conn if conn.try(:bound?)
     end
 
     Rails.logger.error("Unable to bind to any LDAP server: #{CONFIG['ldap_servers']}")
     nil
   end
 
+  def self.ldap_port
```

Comment:
No, I'm fine with testing it in the connection test. However, I think it would be good to test that overwriting the port in the connection test works?